src/js/ripple/orderbook.js

```diff
@@ -1176,4 +1339,347 @@ OrderBook.prototype.is_valid = function() {
+        updatedTakerGets = legOneTakerGetsFunded
+          .subtract(legTwoTakerPaysFunded);
+
+        this.setLegOneTakerGetsFunded(legOneOffer, updatedTakerGets);
+      }
+
+      legTwoPointer++;
+    } else if (legTwoTakerPaysFunded.compareTo(legOneTakerGetsFunded) > 0) {
+      // Leg two in is greater than leg one out. This means we must clamp down
+      // on leg two in
+      var legTwoQuality = this.getOfferQuality(legTwoOffer);
+
+      autobridgedTakerGets = legOneTakerGetsFunded.divide(legTwoQuality);
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      // Update funded amount since leg two offer was not completely consumed
+      legTwoOffer.taker_gets_funded = this.getOfferTakerGetsFunded(legTwoOffer)
+        .subtract(autobridgedTakerGets)
+        .to_text();
+      legTwoOffer.taker_pays_funded = legTwoTakerPaysFunded
+        .subtract(legOneTakerGetsFunded)
+        .to_text();
+
+      legOnePointer++;
+    } else {
+      // Rare case when leg one out and leg two in are equal. Neither leg needs
+      // to be clamped and both offers are fully consumed
+      autobridgedTakerGets = this.getOfferTakerGetsFunded(legTwoOffer);
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      legOnePointer++;
+      legTwoPointer++;
+    }
+
+    var autobridgedOffer = this.formatAutobridgedOffer(
+      autobridgedTakerGets,
+      autobridgedTakerPays
+    );
+    autobridgedOffer.index = legOnePointer + legTwoPointer;
+
+    this._offersAutobridged.push(autobridgedOffer);
+  }
+};
+
+/**
+ * Format an autobridged offer and compute synthetic values (e.g. quality)
+ *
+ * @param [Amount] takerGets
+ * @param [Amount] takerPays
+ *
+ * @return [Object]
+ */
+
+OrderBook.prototype.formatAutobridgedOffer = function(takerGets, takerPays) {
+  assert(takerGets instanceof Amount, 'Autobridged taker gets is invalid');
+  assert(takerPays instanceof Amount, 'Autobridged taker pays is invalid');
+
+  var autobridgedOffer = {};
+  var quality = takerPays.divide(takerGets);
+
+  autobridgedOffer.TakerGets = {
+    value: takerGets.to_text(),
+    currency: this._currencyGets.to_hex(),
+    issuer: this._issuerGets
+  };
+
+  autobridgedOffer.TakerPays = {
+    value: takerPays.to_text(),
+    currency: this._currencyPays.to_hex(),
+    issuer: this._issuerPays
+  };
+
+  autobridgedOffer.quality = quality.to_text();
+
+  autobridgedOffer.taker_gets_funded = autobridgedOffer.TakerGets.value;
+  autobridgedOffer.taker_pays_funded = autobridgedOffer.TakerPays.value;
+
+  autobridgedOffer.autobridged = true;
+
+  autobridgedOffer.BookDirectory = this.convertOfferQualityToHex(quality);
+
+  return autobridgedOffer;
+};
+
+/**
+ * Formats an offer quality amount to a hex that can be parsed by
+ * Amount.parse_quality
+ *
+ * @param [Amount] quality
+ *
+ * @return [String]
+ */
+
+OrderBook.prototype.convertOfferQualityToHex = function(quality) {
```

Comment:
Most of this is copy-pasted from serializedtypes.js, is there a way to use the STAmount class for this?